pluralsight/design-system
# collapsible: ensure it expands completely on page load

### What You're Solving

This hopefully fixes #946. Here's what I wrote at the bottom of that issue:

I'm thinking the main problem here is that when the page loads, the height of the collapsible is just a tiny bit off (could even be a fraction of a pixel) from the actual number it needs to be. So the line that should know that it's already opened is failing:

```javascript
element.style.height === `${element.getBoundingClientRect().height}px`
```

So the code that sets the height back to auto never runs.

The solution in this PR is to wait a maximum of the transition duration time before setting the height back to auto. That would ensure we're always resetting back to auto eventually instead of being stuck at the hard-coded height.

### How to Verify

1. Without my changes, in the Collapsible component, comment out line 27:

```javascript
if (isAlreadyAtTargetHeight) return Promise.resolve()
```

2. Now we need to use collapsible in a way where the Collapsible height changes after it loads. I did that by writing a story in the Drawer's storybook:

```jsx
function DelayedContentLoadStory() {
  const [isOpen, setIsOpen] = useState(true)
  const [contentLines, setContentLines] = useState([1])

  useEffect(() => {
    setTimeout(() => {
      setContentLines([1, 2, 3, 4, 5, 6, 7, 8, 9, 10])
    }, 2000)
  }, [])

  return (
    <div>
      <button type="button" onClick={() => setIsOpen(!isOpen)}>Toggle</button>
      <Collapsible isOpen={isOpen}>
        {contentLines.map(c => (
          <div style={{background: 'blue'}}>{c} of {contentLines.length}</div>
        ))}
      </Collapsible>
    </div>
  )
}
```

3. Verify that 2 seconds after loading the page, the box does not expand to fit the new content.

4. Pull in this code

5. Verify that 2 seconds after loading the page, the box DOES expand to fit the new content.
